### PR TITLE
SurveyPublishJob: UserPrefs set to '0' cycle pref when cycle is among…

### PR DESCRIPTION
**Pull request by fefrain**
… deleted ones

### CODE REVIEW

File: server/modules/user/repository/userRepository.js [hunk 1]
```diff
@@ -126,6 +127,19 @@ const deleteUsersPrefsSurvey = async (surveyId, client = db) => {
 `)
 }
 
+/**
+ * Sets survey cycle user pref to Survey.cycleOneKey if the preferred cycle is among the specified (deleted) ones
+ */
+const deleteUsersPrefsCycleDeleted = async (surveyId, cycleKeysDeleted, client = db) => {
```
Comment (minotogna) [inline server/modules/user/repository/userRepository.js]: Maybe this method should just be ```deleteUsersPrefsSurveyCycle = async (surveyId, cycles, client = db) ```

Comment (fefrain) [reply server/modules/user/repository/userRepository.js]: or even better, resetUsersPrefsSurveyCycle

File: server/modules/survey/service/publish/jobs/userPrefsCycleResetJob.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const R = require('ramda')
+
+const Job = require('../../../../../job/job')
+
+const UserManager = require('../../../../user/manager/userManager')
+const SurveyPublishJobUtils = require('../surveyPublishJobUtils')
+
+class UserPrefsCycleResetJob extends Job {
```
Comment (minotogna) [inline server/modules/survey/service/publish/jobs/userPrefsCycleResetJob.js]: what about one job called ```CyclesDeletedCheckJob``` that removes the records and users prefs?

Comment (fefrain) [reply server/modules/survey/service/publish/jobs/userPrefsCycleResetJob.js]: good idea

**Pull request merged**